lightly/api/api_workflow_datasets.py

```diff
@@ -19,11 +19,32 @@ def _get_current_dataset(self) -> DatasetData:
         """Returns the dataset with id == self.dataset_id.
         
         """
-        dataset_id = self.dataset_id
-        datasets: List[DatasetData] = self._datasets_api.get_datasets()
-        dataset = next(dataset for dataset in datasets if dataset.id == dataset_id)
+        return self.get_dataset_by_id(self.dataset_id)
+
+    def get_dataset_by_id(self, dataset_id: str):
+        """Returns the dataset for the given dataset id. """
+        dataset = self._datasets_api.get_dataset_by_id(dataset_id)
         return dataset
 
+    def get_datasets(self, shared: bool = False) -> List[DatasetData]:
+        """Returns all datasets the user owns.
+
+        Args:
+            shared:
+                If True, only returns the datasets which have been shared with
+                the user.
+
+        """
+        datasets = self._datasets_api.get_datasets(shared=shared)
+        return datasets
+
+    def get_all_datasets(self) -> List[DatasetData]:
+        """Returns all datasets the user has access to. """
```

Comment:
as this function is called quite often (as it seems) to only check if the supplied id the user is passing is actually a valid datasetId, can we cache this information somehow?

Reply:
Let's not add caching just yet as this may lead to many bugs. I will instead change the check to simply request the dataset by id.